src/network/networkd-dhcp4.c

```diff
@@ -94,8 +94,10 @@ static int link_set_dhcp_routes(Link *link) {
                 return log_link_warning_errno(link, r, "DHCP error: could not get address: %m");
 
         r = sd_dhcp_lease_get_router(link->dhcp_lease, &gateway);
-        if (r < 0 && r != -ENODATA)
-                return log_link_warning_errno(link, r, "DHCP error: could not get gateway: %m");
+        if (r == -ENODATA)
+                log_link_info(link, "DHCP: No routes received from DHCP server: %m");
+        else
```

Comment:
Should be `else if (r < 0)` here? With this PR, I got
```
wlp2s0: DHCP error: could not get gateway: Success
```

Reply:
Yes right . updated. thanks !

Reply:
Updated commit fixes my point. Thanks.